pre_commit_hooks/check_executables_have_shebangs.py

```diff
@@ -26,9 +31,9 @@ def check_executables(paths: List[str]) -> int:
 
 
 def _check_git_filemode(paths: Sequence[str]) -> int:
-    outs = cmd_output('git', 'ls-files', '--stage', '--', *paths)
+    outs = cmd_output('git', 'ls-files', '-z', '--stage', '--', *paths)
     seen: Set[str] = set()
-    for out in outs.splitlines():
+    for out in zsplit(outs):
```

Comment:
please add a test for the weird filename as well (make sure it fails without your change too!)

Reply:
done